types/index.d.ts

```diff
@@ -91,6 +91,10 @@ declare module "hyperapp" {
     readonly tag?: Tag<S>
     readonly key: Key
     memo?: PropList<S>
+
+    // `_VDOM` is a guard property which gives us a way to tell `VDOM` objects
+    // apart from `PropList` object.
+    _VDOM: true
```

Comment:
Can we name this guard prop whatever we want? What about `ours`? Bad idea?

Reply:
We can name it whatever we want. I chose `_VDOM` because it seemed the most straightforward thing at the time.

Let's say for the sake of argument we wanted to generalize it so we can use it elsewhere like in `State` or whatever.  We could use something like `_h: "VDOM"` for VDOMs and `_h: "blahbiddyblah"` for other things.

I'm open to the idea but I have yet to think of where else it could be useful and worth the trade-off in type complexity.